internal/bminventory/inventory.go

```diff
@@ -1421,24 +1430,24 @@ func (b *bareMetalInventory) UpdateClusterInstallConfig(ctx context.Context, par
 	if err != nil {
 		log.WithError(err).Errorf("failed to find cluster %s", params.ClusterID)
 		if gorm.IsRecordNotFoundError(err) {
-			return installer.NewUpdateClusterInstallConfigNotFound().WithPayload(common.GenerateError(http.StatusNotFound, err))
+			return nil, common.NewApiError(http.StatusNotFound, err)
 		} else {
-			return installer.NewUpdateClusterInstallConfigInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+			return nil, common.NewApiError(http.StatusInternalServerError, err)
 		}
 	}
 
 	if err = installcfg.ValidateInstallConfigPatch(log, &cluster, params.InstallConfigParams); err != nil {
-		return installer.NewUpdateClusterInstallConfigBadRequest().WithPayload(common.GenerateError(http.StatusBadRequest, err))
+		return &cluster, common.NewApiError(http.StatusBadRequest, err)
```

Comment:
maybe `return nil, common.NewApiError(http.StatusBadRequest, err)`

Reply:
i think it really depends on what we are doing in all other cases but sounds like it is right behavior from golang point of view to return nil